spec/message.abnf

```diff
@@ -0,0 +1,58 @@
+message = [s] *(declaration [s]) body [s]
+
+declaration = let s variable [s] "=" [s] "{" [s] expression [s] "}"
+body = pattern
+     / (selectors 1*([s] variant))
+
+pattern = "{" *(text / placeholder) "}"
+selectors = match 1*([s] selector)
+selector = "{" [s] expression [s] "}"
+variant = when 1*(s key) [s] pattern
+key = nmtoken / literal / "*"
+
+placeholder = "{" [s] (expression / markup / markup-end) [s] "}"
```

Comment:
I think we can fix the markup by moving around some of the constructs. Perhaps:

```abnf
pattern = "{" *(text / placeholder / markup) "}"
markup = markup-start [s] *(text / placeholder) [s] markup-end
placeholder = "{" [s] expression [s] "}"
```

Reply:
This proposal changes three aspects of how markup currently works that would need to be discussed separately:
- It disallows markup within markup, so e.g. `{+b}{+i}...{-i}{-b}` becomes invalid.
- It requires markup start & end tags to be matched within a message.
- It swallows whitespace adjacent to a markup tag.

Reply:
@eemeli 

* It could be fixed to allow embedded markup.
* You're right that it would swallow whitespace and would need fixing that way
* It does require matched markup. We discussed this elsewhere and I filed #356 about that. Given that choice, this would not be appropriate.

So I'll resolve this comment. But `markup` feels unresolved to me. The failure states and fallback states of markup need to be carefully thought through.